conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -62,123 +62,188 @@ object DynamicInterpreter {
+      } else {
+        interpreters
+      }
+    })
+  }
+
+  /**
+    * Returns an interpreter for a conformance rule. Most conformance rules correspond to one interpreter.
+    * The exception is the mapping rule for which there are several interpreters based on the strategy used.
+    *
+    * @param rule   A conformance rule.
+    * @param schema A schema of a DataFrame to be conformed.
+    * @return A conformance rules interpreter.
+    */
+  private def getInterpreter(rule: ConformanceRule,
+                             schema: StructType)
+                            (implicit ictx: InterpreterContext): RuleInterpreter = {
+    rule match {
+      case r: DropConformanceRule             => DropRuleInterpreter(r)
+      case r: ConcatenationConformanceRule    => ConcatenationRuleInterpreter(r)
+      case r: LiteralConformanceRule          => LiteralRuleInterpreter(r)
+      case r: SingleColumnConformanceRule     => SingleColumnRuleInterpreter(r)
+      case r: SparkSessionConfConformanceRule => SparkSessionConfRuleInterpreter(r)
+      case r: UppercaseConformanceRule        => UppercaseRuleInterpreter(r)
+      case r: CastingConformanceRule          => CastingRuleInterpreter(r)
+      case r: NegationConformanceRule         => NegationRuleInterpreter(r)
+      case r: MappingConformanceRule          => getMappingRuleInterpreter(r, schema)
+      case r: CustomConformanceRule           => r.getInterpreter()
+      case r                                  => throw new IllegalStateException(s"Unrecognized rule class: ${r.getClass.getName}")
+    }
+  }
+
+  /**
+    * Returns an interpreter for a mapping rule based on which strategy is applicable.
+    *
+    * @param rule   A conformance rule.
+    * @param schema A schema of a DataFrame to be conformed.
+    * @return A mapping rule interpreter.
+    */
+  private def getMappingRuleInterpreter(rule: MappingConformanceRule,
+                                        schema: StructType)
+                                       (implicit ictx: InterpreterContext): RuleInterpreter = {
+    if (ictx.featureSwitches.experimentalMappingRuleEnabled) {
+      if (canMappingRuleBroadcast(rule, schema)) {
+        MappingRuleInterpreterBroadcast(rule, ictx.conformance)
+      } else {
+        MappingRuleInterpreterGroupExplode(rule, ictx.conformance)
+      }
+    } else {
+      MappingRuleInterpreter(rule, ictx.conformance)
     }
+  }
 
-    (confd, explodeContext)
+  /**
+    * An explosion is needed for a group of mapping rules if the number of mapping rules inside the group
+    * for which broadcasting strategy is not applicable is bigger than 1.
+    *
+    * @param rules  A list of conformance rules grouped by output field being in the same array
+    * @param schema A schema of a dataset
+    * @return true if a group explosion optimization can be used
+    */
+  private def isGroupExploisonCanBeUsed(rules: List[ConformanceRule], schema: StructType): Boolean = {
```

Comment:
```suggestion
  private def isGroupExplosionUsable(rules: List[ConformanceRule], schema: StructType): Boolean = {
```

I think this reads better + typo in Explosion